articles/114_generated_interfaces_python.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: default
+title: Generated Python interfaces
+permalink: articles/generated_interfaces_python.html
+abstract:
+  This article describes the generated Python code for ROS 2 interfaces.
+published: true
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Scope
+
+This article specifies the generated Python code for ROS interface types defined in the [interface definition article](interface_definition.html).
+
+## Namespacing
+
+All code of a ROS package should be defined in a namespace named after the package.
+To separate the generated code from other code within the package it is defined in a sub module:
+
+- namespace for ROS messages: `<package_name>.msg`.
+- namespace for ROS services: `<package_name>.srv`.
+
+## Generated files
+
+Following the Python conventions the namespace hierarchy is mapped to a folder structure.
+The package and module names and therefore the folder and file names use lowercase alphanumeric characters with underscores for separating words (following PEP 8).
+Python files end with `.py`.
+
+## Messages
+
+For a message a Python `class` with the same name is generated in the file `<my_message_name>.py`.
+
+### Types
+
+#### Mapping of primitive types
+
+| ROS type | Python type                  |
+| -------- | ---------------------------- |
+| bool     | builtins.bool                |
+| byte     | builtins.bytes with length 1 |
+| char     | builtins.str with length 1   |
+| float32  | builtins.float               |
+| float64  | builtins.float               |
+| int8     | builtins.int                 |
+| uint8    | builtins.int                 |
+| int16    | builtins.int                 |
+| uint16   | builtins.int                 |
+| int32    | builtins.int                 |
+| uint32   | builtins.int                 |
+| int64    | builtins.int                 |
+| uint64   | builtins.int                 |
+| string   | builtins.str                 |
```

Comment:
I agree that any encoding check should happen in pure Python code. Simply because the goal is that the Python data structure are usable (and consistent) when being used independently.

We are not using Python 3.5 right now since the current version on Trusty is 3.4. But we should keep this in mind when updating the dependencies with Xenial.